### ISSUE TITLE
MuAViC Issue

### ISSUE THREAD

**Issue by aristosp** [roudimit/whisper-flamingo#15]
Hello there,

Seeing as you worked with muavic as well, after seeing your issues at that repo and considering MuAViC repo has been archived, if i may ask the following:
Were you successful in replicating results from that paper? If so, did you make any modifications?
I am trying to replicate results in the German (de) but I receive a 5% higher WER, in clean conditions, my command for inference is:
```
python -B ${AV_HUBERT}/infer_s2s.py \
    --config-dir ${AV_HUBERT}/conf \
    --config-name s2s_decode \
        common.user_dir=${AV_HUBERT} \
        override.modalities=[${MODALITIES}] \
        dataset.gen_subset=${GROUP} \
        override.data=/home_directory/datasets/muavic/de/labels \
        override.label_dir=/home_directory/datasets/muavic/de/labels \
        override.tokenizer_bpe_model=${TOKENIZER_PATH} \
        generation.beam=5 \
        generation.lenpen=0 \
        dataset.max_tokens=3000 \
        override.labels=[${TGT}] \
        override.eval_bleu=${USE_BLEU} \
        common_eval.path=${MODEL_PATH} \
        common_eval.results_path=${OUT_PATH}/${LANG}_${TASK
```
while for training:
```
fairseq-hydra-train \
    --config-dir ${AV_HUBERT}/conf/finetune \
    --config-name base_vox_433h \
        common.user_dir=${AV_HUBERT} \
        task.data=/home_directory/datasets/muavic/de/labels \
        task.modalities=[audio,video] \
        task.label_dir=/home_directory/datasets/muavic/de/labels \
        task.labels=[${TGT}] \
        +task.min_sample_size=5 \
        task.max_sample_size=500 \
        +task.noise_wav=/home_directory/datasets/musan/tsv/all \
		+task.noise_prob=0.25 \
		+task.noise_snr=0  \
        task.tokenizer_bpe_model=${TOKENIZER_PATH} \
        dataset.train_subset=train \
        dataset.valid_subset=valid \
        model.w2v_path=${PRETRAINED_MODEL_PATH} \
        model.freeze_finetune_updates=4000 \
        model.dropout=0.1 \
        +model.no_pretrained_weights=false \
        optimization.lr=[0.0001] \
        optimization.max_update=30000 \
        lr_scheduler.warmup_steps=10000 \
        lr_scheduler.decay_steps=20000 \
        hydra.run.dir=${OUT_PATH}/${LANG}
```

**Comment by aristosp** [comment]
Also, you provide a way to download MuAViC but when I try get 404 error (Not Found), while when I try to access the website from a browser, I get 403 (access denied). Is the dataset not available from you anymore?

**Comment by roudimit** [comment]
Thanks for the message. I updated the download links, these should work:
```
wget https://data.csail.mit.edu/public-release-sls/mwhisper-flamingo/muavic-multi.tar.gz # download all langs
wget https://data.csail.mit.edu/public-release-sls/mwhisper-flamingo/muavic-de.tar.gz # download de
```
Yes, I also found some differences between what I got with their checkpoints and what they reported and it depended on the language, modality, and noise setting. Here were my full results in case they help:

<img width="1456" height="638" alt="Image" src="https://github.com/user-attachments/assets/d6addeee-defc-47f9-9bde-a6e928eafa72" />

<img width="1456" height="638" alt="Image" src="https://github.com/user-attachments/assets/f9b8fff4-c620-4515-a7e0-0a357c40333b" />

**Comment by aristosp** [comment]
Hello and thank you for your comment, my differences are higher, so I guess I'll look into it more. I'll close the issue

**Comment by aristosp** [comment]
Hello again,

I'm reopening this as I haven't been able to reproduce their results. Having said that, may I ask what is the package version for numpy, torch, scipy?  I have tried a variety of versions but still the closest I get is 58% WER. Please let me know if you would like me to close this, as my issue is not specifically to your work, which is great.

**Comment by roudimit** [comment]
I will try to check that for you. Are you able to match the WER for the audio-only case at all? I would start with that, and if you're able to get it to match, then check the video in more detail (ensure that face crops were extracted correctly, and the videos are not too compressed).

**Comment by aristosp** [comment]
I've pinpointed the problem to the fine-tuning process, as I am able to replicate their results, decoding their checkpoint with the same data. However, I do not know what the problem is exactly, as I have not changed anything, besides the gpu number and the update frequency (1 and 8 respectively), using the script shown in the previous comment. I appreciate your time, considering it is not an issue with your work.

**Comment by aristosp** [comment]
In case someone has issues, for me, changing to learning rate = 0.001 (instead of the provided 0.0001 in the train.sh script) seems to lead to results similar to the paper. Merry Christmas and Happy New Year!
